tests/org.eclipse.swt.tests/JUnit Tests/org/eclipse/swt/tests/junit/Test_org_eclipse_swt_widgets_Text.java

```diff
@@ -1669,4 +1666,27 @@ private Event keyEvent(int key, int type, Widget w) {
 	e.widget = w;
 	return e;
 }
+
+/**
+ * Issue 472 - NPE on macOS when creating multiline Text inside Group
+ */
+@Test
+public void test_issue472() {
+	shell.open();
+
+	// On different macOS, different font height is needed to reproduce
+	for (int iFontHeight = 10; iFontHeight < 20; iFontHeight++)
+	{
+		Group group = new Group(shell, SWT.NONE);
+		group.setText("Group");
+		group.setFont(new Font(shell.getDisplay(), "", iFontHeight, 0));
```

Comment:
Font objects need to be disposed, see test failure:

```
Caused by: java.lang.Error: SWT Resource was not properly disposed
	at org.eclipse.swt.graphics.Resource.initNonDisposeTracking(Resource.java:172)
	at org.eclipse.swt.graphics.Resource.<init>(Resource.java:120)
	at org.eclipse.swt.graphics.Font.<init>(Font.java:145)
	at org.eclipse.swt.tests.junit.Test_org_eclipse_swt_widgets_Text.test_issue472(Test_org_eclipse_swt_widgets_Text.java:1682)
```

Reply:
Right, thanks for noticing. I didn't notice because I already got used that some tests always fail.

Reply:
> I didn't notice because I already got used that some tests always fail.

You are not alone, see this lovely discussion at https://github.com/eclipse-platform/eclipse.platform.ui/pull/721#issuecomment-1512818194

@laeubi : this is exactly what the [broken window theory](https://en.wikipedia.org/wiki/Broken_windows_theory) explaims / predicts.

Reply:
@iloveeclipse that becomes completely hilarious, so we have an additional test failure that won't be discovered otherwise but for sure its the fault of the action. So please stop complaining and fix the "broken windows", it wasn't me that "broke" anything or started suggesting to ignoring test.

The github action even showed clearly that there are ADDITIONAL failures, so don't use that as a excuse for not following common committers responsibilities!

Reply:
> So please stop complaining and fix the "broken windows

I personally believe to fix enough, so no need for an extra invite.

> don't use that as a excuse for not following common committers responsibilities!

Have you *tried* to understand https://en.wikipedia.org/wiki/Broken_windows_theory? Please read the full article, not the first few lines.

Reply:
Have you _tried_ to understand that this PR is adding a test for **macOS** that without the additional verification wont be "verified" at all? Have you _tried_ to understand that the action even **has clearly showed** that there was an additional test failure? Has you even _noticed_ that a careful reviewer even could use that information to note that there was something wrong?

> read the full article, not the first few lines.

So maybe even up to this section? 
https://en.wikipedia.org/wiki/Broken_windows_theory#Criticism


So I really don't get why you repeatably reference **me** if _other_ committers **ignore** their responsibility (either by intend or accidentally) so you probably should help them to be more productive instead of referencing the same irrelevant article again and again.

Reply:
Because I hoped you would understand the problem, but I give up now.

Reply:
@SyntevoAlex there are some long pending issues (about a year now) and I think @akurtakov has investigated some things already that there are differences between different linux distribution and/or windowing systems, so yes there is a problem but it seems the test is testing something a user won't notice at all or is irrelevant or no user complained (yet):
- GlyphMetrics: https://github.com/eclipse-platform/eclipse.platform.swt/issues/118
- Browser: https://github.com/eclipse-platform/eclipse.platform.swt/issues/103 

For `GlyphMetrics` I think there are two cases:

1. It diverge between linux systems but is okay, then we probably should simply adjust the test to accept both "good values" here.
2. It is not okay and the code should be adjusted to also work on [ubuntu-latest](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources) (as of today Ubuntu 22.04) as it seems a commonly used system of Eclipse users.

For the Browser thing it seems to be an issue with that the browser modifies the HTML and the test asserts very strict, I proposed using [jsoup](https://jsoup.org/) instead (or one could try "normalize" the html manually) but no one has give any feedback on that idea yet and I don't know enough about the browser test how important the strict assert here is ... and the DND i have no idea at all and can't try it out as I usually not develop with windows.

Reply:
I think that it would be better to make the individual Issues, and probably an umbrella Issue listing unfixed problems in tests, then split your comments here into respective Issues.